UnityWeld/Binding/Internal/TypeResolver.cs

```diff
@@ -310,5 +366,73 @@ public static BindableMember<PropertyInfo>[] FindBindableCollectionProperties(Co
                 .Where(p => !typeof(string).IsAssignableFrom(p.Member.PropertyType))
                 .ToArray();
         }
+
+        /// <summary>
+        /// Register custom binding attribute.
+        /// This will allow to mark bindable properties in external dlls without referencing UnityWeld.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        public static void RegisterCustomBindingAttribute<T>() where T : Attribute

```

Comment:
Since we want `RegisterCustomBindingAttribute` and `RegisterCustomViewModelProvider` to be used externally, it would make more sense to either move this class out of the `UnityWeld.Binding.Internal` namespace or move those functions to a separate class outside that namespace. It occurs to me now that everything inside the `UnityWeld.Binding.Internal` namespace should probably also use the `internal` access modifier instead of `public` but that's not in the scope of this PR so don't worry about changing that.

Reply:
I agree. Was confused by public access of this class.
But I'm not totally sure what the best way to do this...
In my opinion this is strongly relate to IoC principle, so I think I'll made class like "TypeRegistry" inside IoC namespace. And TypeResolver will use public properties of this class.

pseudo:
class TypeRegistry
   BindingAttributeTypes { get; }
   ViewModelProviders { get; }

   RegisterCustomBindingAttribute()
   RegisterCustomViewModelProvider()


Do you agree?